oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyStateTest.java

```diff
@@ -81,4 +110,288 @@ public void multiValuedBinarySize() throws Exception {
+                "tab:a\tb", "nul:a\u0000b"};
+
+        DocumentPropertyState.setCompressionThreshold(1);
+        for (String test : tests) {
+            DocumentPropertyState state = new DocumentPropertyState(store, "propertyName", test, Compression.GZIP);
+            assertEquals(test, state.getValue());
+        }
+    }
+
+    @Test
+    public void testBrokenSurrogateWithoutCompressionForMongo() throws CommitFailedException {
+        getBrokenSurrogateAndInitializeDifferentStores(MONGO, false);
+    }
+
+    @Test
+    public void testBrokenSurrogateWithoutCompressionForRDB() throws CommitFailedException {
+        getBrokenSurrogateAndInitializeDifferentStores(RDB_H2, false);
+    }
+
+    @Test
+    public void testBrokenSurrogateWithoutCompressionForInMemory() throws CommitFailedException {
+        getBrokenSurrogateAndInitializeDifferentStores(MEMORY, false);
+    }
+
+    @Test
+    public void testBrokenSurrogateWithCompressionForMongo() throws CommitFailedException {
+        DocumentPropertyState.setCompressionThreshold(1);
+        getBrokenSurrogateAndInitializeDifferentStores(MONGO, true);
+    }
+
+    @Test
+    public void testBrokenSurrogateWithCompressionForRDB() throws CommitFailedException {
+        DocumentPropertyState.setCompressionThreshold(1);
+        getBrokenSurrogateAndInitializeDifferentStores(RDB_H2, true);
+    }
+
+    @Test
+    public void testBrokenSurrogateWithCompressionForInMemory() throws CommitFailedException {
+        DocumentPropertyState.setCompressionThreshold(1);
+        getBrokenSurrogateAndInitializeDifferentStores(MEMORY, true);
+    }
+
+    private void getBrokenSurrogateAndInitializeDifferentStores(DocumentStoreFixture fixture, boolean compressionEnabled) throws CommitFailedException {
+        String test = "brokensurrogate:dfsa\ud800";
+        DocumentStore store = null;
+        DocumentNodeStore nodeStore = null;
+
+        try {
+            store = fixture.createDocumentStore();
+
+            if (store instanceof MongoDocumentStore) {
+                // Enforce primary read preference, otherwise tests may fail on a
+                // replica set with a read preference configured to secondary.
+                // Revision GC usually runs with a modified range way in the past,
+                // which means changes made it to the secondary, but not in this
+                // test using a virtual clock
+                MongoTestUtils.setReadPreference(store, ReadPreference.primary());
+            }
+            nodeStore = new DocumentMK.Builder().setDocumentStore(store).getNodeStore();
+
+            createPropAndCheckValue(nodeStore, test, compressionEnabled);
+
+        } finally {
+            if (nodeStore != null) {
+                nodeStore.dispose();
+            }
+            if (store != null) {
+                store.dispose();
+            }
```

Comment:
```suggestion
```

`nodeStore.dispose` already disposes the `DocumentStore`, calling it twice here results in a test failure:

```
[ERROR] testBrokenSurrogateWithoutCompressionForRDB(org.apache.jackrabbit.oak.plugins.document.DocumentPropertyStateTest)  Time elapsed: 0.19 s  <<< ERROR!
java.lang.IllegalStateException: Connection handler is already closed (0ms ago)
	at org.apache.jackrabbit.oak.plugins.document.rdb.RDBConnectionHandler.getDataSource(RDBConnectionHandler.java:144)
	at org.apache.jackrabbit.oak.plugins.document.rdb.RDBConnectionHandler.getConnection(RDBConnectionHandler.java:154)
	at org.apache.jackrabbit.oak.plugins.document.rdb.RDBConnectionHandler.getRWConnection(RDBConnectionHandler.java:76)
	at org.apache.jackrabbit.oak.plugins.document.rdb.RDBDocumentStore.dispose(RDBDocumentStore.java:797)
	at org.apache.jackrabbit.oak.plugins.document.DocumentPropertyStateTest.getBrokenSurrogateAndInitializeDifferentStores(DocumentPropertyStateTest.java:308)
```

Reply:
Done.